src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items:T[], key: keyof T):GroupsMap<T> {
+  return items.reduce((accumulator, val) => {
+    const groupedKey = val[key];
+
+    if (!accumulator[groupedKey]) {
+      accumulator[groupedKey] = [];
+    }
+    accumulator[groupedKey].push(val);
+
+    return accumulator;
+  }, {} as any);
```

Comment:
```suggestion
  }, {});
```

Reply:
I wanted to write code  _in style JS_ ! Dont think about TYPE